operations/mimir-mixin/alerts/ingest-storage.libsonnet

```diff
@@ -212,6 +212,36 @@
             message: '%(product)s {{ $labels.%(per_instance_label)s }} in %(alert_aggregation_variables)s Kafka client produce buffer utilization is {{ printf "%%.2f" $value }}%%.' % $._config,
           },
         },
+
+        // Alert if block-builder didn't process cycles in the past hour.
+        {
+          alert: $.alertName('BlockBuilderNoCycleProcessing'),
+          'for': '30m',
+          expr: |||
+            max by(%(alert_aggregation_labels)s) (histogram_count(increase(cortex_blockbuilder_consume_cycle_duration_seconds[1h]))) == 0
```

Comment:
Two comments here:
1. I would suggest alerting on a per-pod basis, so that we notice if a single block-builder is stuck: `max by(cluster, namespace, pod) (histogram_count(increase(cortex_blockbuilder_consume_cycle_duration_seconds[1h]))) == 0`
2. Doing `[1h]` for `30m`, or doing `[90m]` for `1m` is the same thing, so you can easily back test it. Try to run the query `max by(cluster, namespace, pod) (histogram_count(increase(cortex_blockbuilder_consume_cycle_duration_seconds[90m]))) == 0` both in dev and prod. You will see some single replicas stuck from time to time. I would like you to investigate the root cause before adding this alert, because if they're false positives we should find a way to fix the false positives, otherwise we should investigate the issue